frontend/src/components/profile/EditProfile.vue

```diff
@@ -136,102 +147,15 @@ export default {
-                this.updateLastName(value);
-            }
-        },
-        phone: {
-            get () {
-                return this.user.phone_number;
-            },
-            set (value) {
-                this.updatePhone(value);
-            }
-        },
-        facebook: {
-            get () {
-                return this.user.facebook_url;
-            },
-            set (value) {
-                this.updateFacebook(value);
-            }
-        },
-        instagram: {
-            get () {
-                return this.user.instagram_url;
-            },
-            set (value) {
-                this.updateInstagram(value);
-            }
-        },
-        twitter: {
-            get () {
-                return this.user.twitter_url;
-            },
-            set (value) {
-                this.updateTwitter(value);
-            }
-        }
     },
     methods: {
         ...mapActions({
             updateProfile: 'profile/updateProfile',
-            deleteUserAvatar: 'profile/deleteUserAvatar',
-            setUserAvatar: 'profile/setUserAvatar',
-            setUser: 'profile/setUser',
-
-            updateFirstName: 'profile/updateFirstName',
-            updateLastName: 'profile/updateLastName',
-            updatePhone: 'profile/updatePhone',
-            updateFacebook: 'profile/updateFacebook',
-            updateInstagram: 'profile/updateInstagram',
-            updateTwitter: 'profile/updateTwitter',
-            updateDateOfBirth: 'profile/updateDateOfBirth',
-            updateUserAvatar: 'profile/updateUserAvatar',
-            updatePassword: 'profile/updatePassword',
+            deleteUserAvatar: 'profile/deleteUserAvatar'
         }),
         saveData() {
-            if (this.oldPassword !== '' && this.newPassword !== '') {
-                this.updatePassword({old_password: this.oldPassword, new_password: this.newPassword})
-                    .then(() => {
-                        this.saveProfileInfo();
-                    })
-                    .catch((data) => {
-                        let message = '';
-                        if (data.error !== undefined) {
-                            message = data.error.message;
-                        } else if (data.errors !== undefined) {
-                            for (let error in data.errors) {
-                                data.errors[error].forEach(function(item) {
-                                    message += item + '<br>';
-                                });
-                            }
-                        }
-                        this.onError({
-                            message: message
-                        });
-                        return false;
-                    });
-            } else {
-                this.saveProfileInfo();
-            }
-        },
-        saveProfileInfo() {
             const formData = new FormData();
             formData.append('first_name', this.user.first_name);
```

Comment:
I think we can make it likein own method
`const formData = this.createFormData()`